src/Symfony/Component/HttpFoundation/JsonResponse.php

```diff
@@ -171,6 +171,11 @@ public function setEncodingOptions($encodingOptions)
     {
         $this->encodingOptions = (int) $encodingOptions;
 
+        // If callback is not used, try to update $this->data with decoded content
+        if ($this->callback === null) {
+            return $this->setData(json_decode($this->content));
```

Comment:
This side effect is unexpected I think. It creates a correlation between 3 properties and adds "magic to the process, and overhead.